### PR TITLE
 [baxtereus] add ik check test for apc

### PR DESCRIPTION
**Pull request by knorth55**
test-ik-all-bin
testing ik for apc kiva pod

### CODE REVIEW

**Comment by k-okada** [comment]
Thanks, so what is the expected behavior? do you want to solve all ik by both hands? if so does the real robot can physically reach to that position?

**Comment by knorth55** [comment]
I tried to solve all ik by both hand, but  the real robot cannot physically reach to the position by both arm.  Actually some position cannot be reached by either arm.
Should I modify all positions to ik solvable positions?

**Comment by k-okada** [comment]
No you do not have to modify the positions, please give us the list of the
positions/bins that the real robot can physically reach.

--
◉ Kei Okada

**Comment by knorth55** [comment]
now it's not WIP

**Comment by knorth55** [comment]
https://github.com/knorth55/jsk_robot/blob/ik-test/jsk_baxter_robot/baxtereus/test/test-baxter.l#L45

x座標方向にそれぞれのbinに対して上記のlistのような制限を設けて、どちらかの腕でikが解ければOKという形でtestを作りました。
手がこれより棚の奥に入った位置ではikが解けないということです。
結果として実機の可動域と比較してもそこまで大きな違いはないと思います。

**Comment by knorth55** [comment]
@k-okada ping

**Comment by knorth55** [comment]
@k-okada please merge this

**Comment by knorth55** [comment]
@k-okada please merge this :)

**Comment by knorth55** [comment]
@k-okada please merge this

**Comment by knorth55** [comment]
@k-okada Please merge this!!!!

File: jsk_baxter_robot/baxtereus/test/test-baxter.l [hunk 1]
```diff
@@ -69,6 +87,108 @@
...
+        ((> tmp-x (+ x (/ dx 2))) t)
+        (do ((tmp-y (+ (- y (/ dy 2)) (/ step-dist 2)) (+ tmp-y step-dist)))
+          ((> tmp-y (+ y (/ dy 2))) t)
+          (do ((tmp-z (+ (- z (/ dz 2)) (/ step-dist 2)) (+ tmp-z step-dist)))
+            ((> tmp-z (+ z (/ dz 2))) t)
+            (setq pos (float-vector tmp-x tmp-y tmp-z))
+            (assert
+              (or
+                (send robot :rarm :inverse-kinematics
+                  (make-cascoords :pos (v+ pos #f(-90 0 0)))
```
Comment (k-okada) [inline jsk_baxter_robot/baxtereus/test/test-baxter.l]: @knorth55 why you added `#f(-90 0 0))` at here ?

Comment (knorth55) [inline jsk_baxter_robot/baxtereus/test/test-baxter.l]: @k-okada 
i followed `deftest test-robot-ik-bin` 
https://github.com/jsk-ros-pkg/jsk_robot/blob/master/jsk_baxter_robot/baxtereus/test/test-baxter.l#L37
i think it's unnecessary now.

File: jsk_baxter_robot/baxtereus/test/test-baxter.l [hunk 2]
```diff
@@ -69,6 +87,108 @@
...
+                  :rotation-axis t
+                  :debug-view :no-message)
+                (send robot :larm :inverse-kinematics
+                  (make-cascoords :pos (v+ pos #f(-90 0 0)))
+                  :rotation-axis t
+                  :debug-view :no-message)
+                (> tmp-x limit))
```
Comment (k-okada) [inline jsk_baxter_robot/baxtereus/test/test-baxter.l]: why you only check the lower limit of the position? you do not need something like `(< tmp-x limit2)` ??

Comment (knorth55) [inline jsk_baxter_robot/baxtereus/test/test-baxter.l]: i checked it with bin and it dont need lower limit

**Comment by wkentaro** [comment]
This test takes too much time, doesn't it? (> 10min)
Could you please simplify it? 
screen shot 2016-05-13 at 10 54 54

**Pull request merged**